apps/prairielearn/src/models/course-instance-usages.sql

```diff
@@ -0,0 +1,70 @@
+-- BLOCK update_course_instance_usages_for_submission
+INSERT INTO
+  course_instance_usages (
+    type,
+    institution_id,
+    course_id,
+    course_instance_id,
+    date,
+    user_id,
+    include_in_statistics
+  )
+SELECT
+  'Submission',
+  i.id,
+  c.id,
+  ci.id,
+  date_trunc('day', s.date, 'UTC'),
+  v.authn_user_id,
```

Comment:
I'm using `v.authn_user_id` for the `user_id` in all cases (submissions, external grading, workspaces). My thinking is that want:
1. The `user_id` must always exist.
2. If an instructor is emulating a student, we want to get the `user_id` of the student. I think that instructors should generally not be generating new variants while emulating.

I considered `gj.auth_user_id`, `s.auth_user_id`, `ai.user_id`, etc, but they all seem to have problems for either criteria (1) or (2) above.

Does this sound correct?

Reply:
Actually I just checked and we have variants with a NULL value for `authn_user_id`. I think these might only be historical, because the current code shouldn't be able to have NULL: https://github.com/PrairieLearn/PrairieLearn/blob/981793c200ca41226dec9e589f757724e0d36f7d/apps/prairielearn/src/lib/question-variant.ts#L216